In [1]:
import pandas as pd
import os

Loading the datset BristolBay_FeatureSet_For_Model.csv

In [2]:
data_set_bristol = pd.read_csv(r'C:\Users\MuriloFarias\Desktop\NNS-JULIA\PredictSalmonRuns\data\BristolBay_FeatureSet_For_Model.csv')

Data Exploratory

Checking the shape

In [3]:
data_set_bristol.shape

(488, 34)

See the data

In [4]:
data_set_bristol

River  Year  Total_Returns  AgeClass_0.1  AgeClass_0.2  AgeClass_0.3  \
0    Alagnak  1963   3.781383e+05      0.000000      0.000000  0.000000e+00   
1    Alagnak  1964   7.593831e+05      0.000000      0.000140  0.000000e+00   
2    Alagnak  1965   3.978606e+05      0.000000      0.000000  7.270738e+02   
3    Alagnak  1966   3.940772e+05      0.000000      0.000000  1.832636e-09   
4    Alagnak  1967   3.443056e+05      0.000000      0.000000  1.073499e-09   
..       ...   ...            ...           ...           ...           ...   
483     Wood  2019   1.234074e+07      0.000000      0.000000  0.000000e+00   
484     Wood  2020   8.004915e+06      0.000000      0.000000  0.000000e+00   
485     Wood  2021   1.267032e+07      0.000001      0.000001  1.052336e-06   
486     Wood  2022   1.241018e+07      0.000001      0.000001  1.795169e-03   
487     Wood  2023   1.041894e+07      0.000001      0.000001  1.016114e-06   

     AgeClass_0.4  AgeClass_0.5  AgeClass_1.1  AgeClass_1.2  ...  \
0        0.000000      0.000000      0.000000  3.313000e+05  ...   
1        0.000000      0.000000   4468.518531  1.213613e+05  ...   
2        0.000000      0.000000  11965.313972  5.029325e+04  ...   
3        0.000000      0.000000      0.000000  1.102662e+05  ...   
4        0.000000      0.000000   6377.297622  2.310019e+05  ...   
..            ...           ...           ...           ...  ...   
483      0.000000      0.000000  87517.701509  8.064807e+06  ...   
484      0.000000      0.000000  36289.024162  5.724076e+06  ...   
485      0.000001      0.000001   8527.016450  1.113983e+07  ...   
486      0.000001      0.000001   7208.086447  7.504564e+06  ...   
487      0.000001      0.000001  61995.910249  3.376879e+06  ...   

     oni_mean_DecFeb  mei_mean_AprSep  npgo_mean_DecFeb  ao_mean_DecMar  \
0          26.003333              NaN         -0.209223        -1.25475   
1          27.330000              NaN          0.635470        -0.48150   
2          25.780000              NaN         -0.708755        -1.07025   
3          27.813333              NaN         -0.876998        -1.35450   
4          26.073333              NaN         -0.630369         0.29250   
..               ...              ...               ...             ...   
483        27.470000         0.276667         -1.539047         0.66550   
484        27.220000        -0.685000         -2.187605         2.22225   
485        25.653333        -1.196667         -1.064231        -0.82550   
486        25.746667        -1.798333         -1.865551         0.72375   
487        26.063333         0.281667         -1.972965        -0.37825   

     pdo_mean_DecMar  pdo_mean_MaySep  sst_aprjul  sst_anom  sss_mayaug  \
0            -0.0600           -0.580         NaN       NaN   31.649451   
1             0.0725           -1.456         NaN       NaN   31.806597   
2            -0.5875           -0.202         NaN       NaN   31.792252   
3            -0.4625           -0.258         NaN       NaN   31.768258   
4            -0.3750           -1.220         NaN       NaN   31.952063   
..               ...              ...         ...       ...         ...   
483          -0.4175            0.166    7.794615  2.187847   31.467492   
484          -1.1600           -0.914    6.697374  1.090606   31.716720   
485          -1.0900           -1.754    5.850938  0.244170   31.832995   
486          -2.1725           -2.174    5.287018 -0.319750   31.794433   
487          -1.8875           -2.586    4.550619 -1.056149   31.933957   

     Total_Spawners_BroodYear  
0                8.363766e+07  
1                9.066752e+07  
2                5.119336e+08  
3                1.393965e+08  
4                1.071686e+08  
..                        ...  
483              1.178863e+08  
484              1.029989e+08  
485              6.813127e+07  
486              1.784546e+07  
487                       NaN  

[488 rows x 34 columns]

Sum of all columns

In [23]:
# Select only numeric columns (automatically excludes strings like 'River')
numeric_sums = data_set_bristol.select_dtypes(include='number').sum()

# Print with commas and 2 decimals
for col, val in numeric_sums.items():
    print(f"{col}: {val:,.2f}")


Year: 972,584.00
Total_Returns: 2,207,478,149.46
AgeClass_0.1: 2,350.03
AgeClass_0.2: 1,474,512.92
AgeClass_0.3: 8,772,787.70
AgeClass_0.4: 1,247,212.88
AgeClass_0.5: 2,763.06
AgeClass_1.1: 5,610,366.63
AgeClass_1.2: 612,845,787.17
AgeClass_1.3: 668,785,921.37
AgeClass_1.4: 8,217,692.06
AgeClass_1.5: 16,687.73
AgeClass_2.1: 7,994,539.46
AgeClass_2.2: 638,032,541.30
AgeClass_2.3: 248,856,997.78
AgeClass_2.4: 780,826.86
AgeClass_3.1: 132,687.41
AgeClass_3.2: 3,523,385.09
AgeClass_3.3: 1,180,488.40
AgeClass_3.4: 601.61
Total_Returns_NextYear: 2,251,595,670.85
Pacea_ALPI_Anomaly: 206.83
npi_mean_NovMar: 492,393.60
oni_mean_DecFeb: 12,977.12
npgo_mean_DecFeb: -78.14
ao_mean_DecMar: -67.12
pdo_mean_DecMar: -134.16
pdo_mean_MaySep: -207.39
sss_mayaug: 15,441.83
Total_Spawners_BroodYear: 102,375,740,733.92


Check the type of each columns

In [6]:
data_set_bristol.dtypes


River                        object
Year                          int64
Total_Returns               float64
AgeClass_0.1                float64
AgeClass_0.2                float64
AgeClass_0.3                float64
AgeClass_0.4                float64
AgeClass_0.5                float64
AgeClass_1.1                float64
AgeClass_1.2                float64
AgeClass_1.3                float64
AgeClass_1.4                float64
AgeClass_1.5                float64
AgeClass_2.1                float64
AgeClass_2.2                float64
AgeClass_2.3                float64
AgeClass_2.4                float64
AgeClass_3.1                float64
AgeClass_3.2                float64
AgeClass_3.3                float64
AgeClass_3.4                float64
Total_Returns_NextYear      float64
Pacea_ALPI_Anomaly          float64
npi_mean_NovMar             float64
oni_mean_DecFeb             float64
mei_mean_AprSep             float64
npgo_mean_DecFeb            float64
ao_mean_DecMar              

Check all river's name

In [7]:
data_set_bristol['River'].unique()

array(['Alagnak', 'Egegik', 'Igushik', 'Kvichak', 'Naknek', 'Nushagak',
       'Ugashik', 'Wood'], dtype=object)

check missinvalues

In [8]:
data_set_bristol.isnull().sum()


River                         0
Year                          0
Total_Returns                 0
AgeClass_0.1                  0
AgeClass_0.2                  0
AgeClass_0.3                  0
AgeClass_0.4                  0
AgeClass_0.5                  0
AgeClass_1.1                  0
AgeClass_1.2                  0
AgeClass_1.3                  0
AgeClass_1.4                  0
AgeClass_1.5                  0
AgeClass_2.1                  0
AgeClass_2.2                  0
AgeClass_2.3                  0
AgeClass_2.4                  0
AgeClass_3.1                  0
AgeClass_3.2                  0
AgeClass_3.3                  0
AgeClass_3.4                  0
Total_Returns_NextYear        0
Pacea_ALPI_Anomaly            8
npi_mean_NovMar               0
oni_mean_DecFeb               0
mei_mean_AprSep             128
npgo_mean_DecFeb              0
ao_mean_DecMar                0
pdo_mean_DecMar               0
pdo_mean_MaySep               0
sst_aprjul                  168
sst_anom

droping three columns because they have a lot of missvalues
mei_mean_AprSep             128
sst_aprjul                  168
sst_anom                    168

In [9]:
data_set_bristol = data_set_bristol.drop(columns=['sst_anom', 'sst_aprjul', 'mei_mean_AprSep'])


Fill out missvalues in this two columns using average
Pacea_ALPI_Anomaly            8
Total_Spawners_BroodYear      8


In [10]:
data_set_bristol['Pacea_ALPI_Anomaly'].fillna(data_set_bristol['Pacea_ALPI_Anomaly'].mean(), inplace=True)
data_set_bristol['Total_Spawners_BroodYear'].fillna(data_set_bristol['Total_Spawners_BroodYear'].mean(), inplace=True)


C:\Users\MuriloFarias\AppData\Local\Temp\ipykernel_16184\438147762.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data_set_bristol['Pacea_ALPI_Anomaly'].fillna(data_set_bristol['Pacea_ALPI_Anomaly'].mean(), inplace=True)
C:\Users\MuriloFarias\AppData\Local\Temp\ipykernel_16184\438147762.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate ob

Check again after some process to check if the data its done for one hot encode

In [11]:
data_set_bristol.isnull().sum()


River                       0
Year                        0
Total_Returns               0
AgeClass_0.1                0
AgeClass_0.2                0
AgeClass_0.3                0
AgeClass_0.4                0
AgeClass_0.5                0
AgeClass_1.1                0
AgeClass_1.2                0
AgeClass_1.3                0
AgeClass_1.4                0
AgeClass_1.5                0
AgeClass_2.1                0
AgeClass_2.2                0
AgeClass_2.3                0
AgeClass_2.4                0
AgeClass_3.1                0
AgeClass_3.2                0
AgeClass_3.3                0
AgeClass_3.4                0
Total_Returns_NextYear      0
Pacea_ALPI_Anomaly          0
npi_mean_NovMar             0
oni_mean_DecFeb             0
npgo_mean_DecFeb            0
ao_mean_DecMar              0
pdo_mean_DecMar             0
pdo_mean_MaySep             0
sss_mayaug                  0
Total_Spawners_BroodYear    0
dtype: int64

One-Hot Encode River Column

In [12]:
# One-hot encode and convert True/False to 1/0
data_set_bristol_encoded = pd.get_dummies(data_set_bristol, columns=['River'], prefix='River')



In [13]:
river_cols = [col for col in data_set_bristol_encoded.columns if col.startswith('River_')]
data_set_bristol_encoded[river_cols] = data_set_bristol_encoded[river_cols].astype(int)

checking the new columns after One-hot 

In [15]:
data_set_bristol_encoded.columns

Index(['Year', 'Total_Returns', 'AgeClass_0.1', 'AgeClass_0.2', 'AgeClass_0.3',
       'AgeClass_0.4', 'AgeClass_0.5', 'AgeClass_1.1', 'AgeClass_1.2',
       'AgeClass_1.3', 'AgeClass_1.4', 'AgeClass_1.5', 'AgeClass_2.1',
       'AgeClass_2.2', 'AgeClass_2.3', 'AgeClass_2.4', 'AgeClass_3.1',
       'AgeClass_3.2', 'AgeClass_3.3', 'AgeClass_3.4',
       'Total_Returns_NextYear', 'Pacea_ALPI_Anomaly', 'npi_mean_NovMar',
       'oni_mean_DecFeb', 'npgo_mean_DecFeb', 'ao_mean_DecMar',
       'pdo_mean_DecMar', 'pdo_mean_MaySep', 'sss_mayaug',
       'Total_Spawners_BroodYear', 'River_Alagnak', 'River_Egegik',
       'River_Igushik', 'River_Kvichak', 'River_Naknek', 'River_Nushagak',
       'River_Ugashik', 'River_Wood'],
      dtype='object')

Checking if the main values still ok after one hot encode

In [17]:
#original data data_set_bristal
total_return_bristol = data_set_bristol['Total_Returns'].sum()
total_next_bristol = data_set_bristol['Total_Returns_NextYear'].sum()

# after one hot encoding data_set_bristol_encoded
total_return_encoded = data_set_bristol_encoded['Total_Returns'].sum()
total_next_encoded = data_set_bristol_encoded['Total_Returns_NextYear'].sum()

# Differences
diff_return = total_return_encoded - total_return_bristol
diff_next = total_next_encoded - total_next_bristol

print("  Difference in Total_Return:", diff_return)
print("  Difference in Total_Returns_NextYear:", diff_next)

  Difference in Total_Return: 0.0
  Difference in Total_Returns_NextYear: 0.0


changing pd options to show all columns and all samples

In [ ]:
# Show all rows and columns
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
data_set_bristol_encoded


In [ ]:
sums = data_set_bristol_encoded.sum()

# Print each column sum with comma and no scientific notation
for col, val in sums.items():
    print(f"{col}: {val:,.2f}")


In [24]:
data_set_bristol_encoded.shape

(488, 38)